### DISCUSSION TITLE
[Feature Request] Improvement for automatic base change

### DISCUSSION THREAD

**Discussion by akaoss** [community/community#125229] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

Imagine, you have 
- PR#1  branch_a to master with commit1, commit2, commit3
- PR#2 branch_b to branch_a with commit 4

When you merge PR#1 using "Squash and Merge" button, the base branch for PR#2 is automatically changed from branch_a to master, but the list of commits is changed too: from commit4 to commi1+commit2+commit3+commit4. And the list of requested reviewers also includes all of the parties from both PRs.

I'd like github to be smart enough to detect that the merge commit to master is equal to commit1+commit2+commit3 and **keep only commit4 and reviewers needed only for that commit**

**Comment by mirekondro** [comment]
same
